lib/puppet_x/templates/invoke_ps_command.erb

```diff
@@ -1,15 +1,207 @@
 $hostSource = @"
 using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Globalization;
+using System.Management.Automation;
 using System.Management.Automation.Host;
+using System.Security;
+using System.Text;
 using System.Threading;
 
-public class PuppetPSHost : PSHost
+namespace Puppet
 {
+  public class PowerShellRawUserInterface : PSHostRawUserInterface
+  {
+    public PowerShellRawUserInterface()
+    {
+      BufferSize = new Size(80, 25);
+      BackgroundColor = ConsoleColor.Black;
+      ForegroundColor = ConsoleColor.White;
+      CursorPosition = new Coordinates(0, 0);
+      CursorSize = 1;
+    }
+
+    public override ConsoleColor BackgroundColor { get; set; }
+    public override Size BufferSize { get; set; }
+    public override Coordinates CursorPosition { get; set; }
+    public override int CursorSize { get; set; }
+    public override ConsoleColor ForegroundColor { get; set; }
+    public override Coordinates WindowPosition { get; set; }
+    public override Size WindowSize { get; set; }
+    public override string WindowTitle { get; set; }
+
+    public override bool KeyAvailable
+    {
+        get { return Console.KeyAvailable; }
+    }
+    public override Size MaxPhysicalWindowSize
+    {
+        get { return new Size(Console.LargestWindowWidth, Console.LargestWindowHeight); }
+    }
+
+    public override Size MaxWindowSize
+    {
+        get { return new Size(Console.LargestWindowWidth, Console.LargestWindowHeight); }
```

Comment:
This entire class was added only to get the default PowerShell formatting to work, its not usually required by default. We don't actually use the console, just attempt to access the constants.